sigstore/_internal/trustroot.py

```diff
@@ -58,6 +62,19 @@ def _is_timerange_valid(period: TimeRange | None, *, allow_expired: bool) -> boo
     return allow_expired or (period.end is None or now <= period.end)
 
 
+class KeyringPurpose(str, Enum):
+    """
+    Keyring purpose typing
+    """
+
+    SIGN = "Signing"
+    VERIFY = "Verifying"
```

Comment:
Nitpick: either all-lower or all-upper enum values are idiomatic:

```suggestion
    SIGN = "sign"
    VERIFY = "verify"
```